docs/deployment/run-in-prod.md

```diff
@@ -0,0 +1,212 @@
+Uvicorn includes a gunicorn worker class that means you can get set up with very little configuration.
+
+!!! note
+    Gunicorn is mainly an application server using the WSGI standard and not ASGI standard.But Gunicorn supports working as a process manager and allowing users to tell it which specific worker process class to use. Then Gunicorn would start one or more worker processes using that class.
+
+Now we will see how we can deploy using the Gunicorn.
+
+Let's first install Gunicorn.
+```commandline
+pip install gunicorn
+```
+
+Then, to start the Gunicorn with Uvicorn workers
+```commandline
+gunicorn -w 4 -k uvicorn.workers.UvicornWorker
+```
+The above command will start Gunicorn with 4 workers! 
+
+`uvicorn.workers.UvicornWorker` is the worker class we discussed about.
+The `UvicornWorker` implementation uses the `uvloop` and `httptools` implementations for improved performance. 
+
+To run under `PyPy` you'll want to use pure-python implementation instead. You can do this by using the `UvicornH11Worker` class.
+
+If you need to pass Uvicorn's config arguments to gunicorn workers then you'll have to subclass `UvicornWorker`.
+```python
+from uvicorn.workers import UvicornWorker
+
+class MyUvicornWorker(UvicornWorker):
+    CONFIG_KWARGS = {"loop": "asyncio", "http": "h11", "lifespan": "off"}
+```
+
+### Using Supervisor
+
+[Supervisor](http://supervisord.org/) is a client/server system that allows its users to monitor and control a number of processes on UNIX-like operating systems.
+
+To use `supervisor` as a process manager you should either:
+
+* Hand over the socket to uvicorn using its file descriptor, which supervisor always makes available as `0`, and which must be set in the `fcgi-program` section.
+* Or use a UNIX domain socket for each `uvicorn` process.
+
+A simple supervisor configuration might look something like this:
+
+**supervisord.conf**:
+
+```ini
+[supervisord]
+
+[fcgi-program:uvicorn]
+socket=tcp://localhost:8000
+command=venv/bin/uvicorn --fd 0 example:App
+numprocs=4
+process_name=uvicorn-%(process_num)d
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+```
+
+Then run with `supervisord -n`.
+
+### Using Circus
+
+[Circus](https://circus.readthedocs.io/en/latest/) is a Python program which can be used to monitor and control processes and sockets.
+
+To use `circus` as a process manager, you should either:
+
+* Hand over the socket to uvicorn using its file descriptor, which circus makes available as `$(circus.sockets.web)`.
+* Or use a UNIX domain socket for each `uvicorn` process.
+
+A simple circus configuration might look something like this:
+
+**circus.ini**:
+
+```ini
+[watcher:web]
+cmd = venv/bin/uvicorn --fd $(circus.sockets.web) example:App
+use_sockets = True
+numprocesses = 4
+
+[socket:web]
+host = 0.0.0.0
+port = 8000
+```
+
+Then run `circusd circus.ini`.
+
+## Run using Docker Containers
```

Comment:
I don't think we need to dive into Docker here in this much depth. That goes a bit beyond an ASGI server, and there's plenty of content out there on it.

Reply:
I think it’s fare to include the recommended Docker image and how to build and run it. I kept it really short.  We can trim further…

Reply:
Agree with you, I think we can trim all the docker explanation and stick to "how to" alone. The is just the sample `Dockerfile`.

Reply:
I will make the changes and push a version today  :)